TemporaryParser/TPRBParser.class.st

```diff
@@ -851,17 +865,28 @@ TPRBParser >> parseStatementList: pragmaBoolean into: sequenceNode [
 									value: self parseAssignment.
 						statements add: node.
 						return := true]
-					ifFalse: [node := self parseAssignment.
-								statements add: node ].
+					ifFalse: [(currentToken isSpecial and: ['])}' includes: currentToken value]) ifFalse: [node := self parseAssignment.
+								statements add: node ]].
 				(currentToken isSpecial and: [currentToken value = $.]) 
 					ifTrue: 
 						[periods add: currentToken start.
 						self step.
 						self addCommentsTo: node]
-					ifFalse: [return := true].
+					ifFalse: [(currentToken isSpecial and: ['])}' includes: currentToken value])
+									ifTrue: [ statements add: (currentToken value = ']' 
+										ifTrue: [ errorBlock 
+											ifNil: [ self parserError: 'opening expected before closure ', currentToken value]
+					  						ifNotNil: [ | errorNode | errorNode := self parseEnglobingError: statements with: currentToken. statements := OrderedCollection new. errorNode] ]
+										ifFalse: [ errorBlock 
+											ifNil: [ self parserError: 'opening expected before closure', currentToken value]
+					  				  		ifNotNil: [ statements isEmpty 
+												ifFalse: [| errorNode | errorNode := self parseEnglobingError: (OrderedCollection with:statements last) with: currentToken. statements removeLast. errorNode] 
+												ifTrue: [ self parseEnglobingError: statements with: currentToken] ] ]). 
+										self step.]
+									ifFalse: [return := true]].

```

Comment:
This is super messy. Maybe we should cut it in smaller methods? The pattern`errorBlock ifNil: [ self parserError: ] ifNotNil: [  self parseEnglobingError: ]` is back here twice in the same method!